### DISCUSSION TITLE
TextureButton Theming

### DISCUSSION THREAD

**Discussion by Jalkhov** [godotengine/godot-proposals#9897] [GUI]
Hello everyone...

These days I realized that in the theme editor does not appear to add the TextureButton as an item, investigating, the only related thing I found was godotengine/godot#39344, and I would like to know why this is not possible. I will explain the need that came to me and you will say if there is another way to achieve it, I must clarify that I am not an expert in the engine.

What I wanted to do was: Inside a `GridContainer` place some images, and the user could select some of these, and I thought that the `TextureButton` with Toggle mode and a Group came as a ring to my finger, the need comes when I wanted to give a different appearance to the button when it is selected and when it is hovered, and the `TextureButton` was not in the list of items, is there any way to achieve this? And if so, why it is not possible to do it from the theme editor?

**Comment by theraot** [comment]
My guess for the reason why `TextureButton` does not have theme properties is that you can use `StyleBoxTexture` with a regular buttton.

I do not understand why you need `TextureButton` for your use case (note: `button_group` and `toggle_mode` are members of [`BaseButton`](https://docs.godotengine.org/en/stable/classes/class_basebutton.html), so [`Button`](https://docs.godotengine.org/en/stable/classes/class_button.html#class-button) and [`LinkButton`](https://docs.godotengine.org/en/stable/classes/class_linkbutton.html#class-linkbutton) also have them).

Thus, I'd argue you should use `Button`.

By the way, if you already have styling for `Button` in your `Theme`, know that you can have multiple styles for `Button` on the same theme using [Theme type variations](https://docs.godotengine.org/en/stable/tutorials/ui/gui_theme_type_variations.html).

And if that is not enough to get the style you want, know that a `Button` can have children, so you could have a `Panel` or `PanelContainer` and a `Lebel` (so you can use `LabelSettings`) or `RichTextLabel` (So you can use `BBCode`) inside the `Button` (you can leave the `text` empty, and use the `Lebel` or `RichTextLabel` inside).

**Reply by Jalkhov** [reply]
I see, thank you very much for the guidance, as I said I am not an expert on the engine, but I did consider the idea of assigning a child to a Button, I just wanted to get out of the doubt of why I could not and have another point of view, in the end my solution (for now) was:

```
Button
└─ TextureRect
```
Thank you very much for the guidance again.
